src/test/signin.test.js

```diff
@@ -0,0 +1,244 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import bcrypt from 'bcrypt';
+
+import models from '../db/models';
+import app from '../index';
+
+chai.use(chaiHttp);
+
+const { expect } = chai;
+const signinUrl = '/api/v1/auth/signin';
+const { User } = models;
+
+const testUserDetails = {
+  first_name: 'Olawumi',
+  last_name: 'Yusuff',
+  email: 'user@test.com',
+  password: '12345678'
+};
+let testUser;
+
+
+before((done) => {
+  bcrypt.hash(testUserDetails.password, 10)
+    .then((hash) => User.create({
+      first_name: testUserDetails.first_name,
+      last_name: testUserDetails.last_name,
+      email: testUserDetails.email,
+      password: hash
+    }))
+    .then((user) => {
+      testUser = user;
+      done();
+    })
+    .catch((e) => done(e));
+});
+
+after((done) => {
+  User.destroy({
+    where: {
+      email: testUserDetails.email
+    }
+  })
+    .then(() => done())
+    .catch((e) => done(e));
+});
+
+describe(`POST ${signinUrl}`, () => {
+  describe('SUCCESS', () => {
+    it('should sign in a user successfully', async () => {
+      const res = await chai.request(app)
+        .post(signinUrl)
+        .send({
+          email: testUserDetails.email,
+          password: testUserDetails.password
+        });
+
+      expect(res.status).to.equal(200);
```

Comment:
Instead of checking properties one after the other, you can do this
```
expect(res.body.data).to.have.keys('a', 'b'....)

```

Great work though.

Reply:
Cool, I didn't know that. Thanks.